libpromises/generic_agent.c

```diff
@@ -1591,7 +1591,17 @@ void GenericAgentInitialize(EvalContext *ctx, GenericAgentConfig *config)
 
     EvalContextClassPutHard(ctx, "any", "source=agent");
 
-    GenericAgentAddEditionClasses(ctx);
+    GenericAgentAddEditionClasses(ctx); // May set "enterprise_edition" class
+
+    const Class *enterprise_edition = EvalContextClassGet(ctx, "default", "enterprise_edition");
+    if (enterprise_edition == NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "cf_edition", "community", CF_DATA_TYPE_STRING, "derived-from=enterprise_edition,report");
```

Comment:
I see we have one more tag which we typically use for things like this:

```suggestion
        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "cf_edition", "community", CF_DATA_TYPE_STRING, "derived-from=enterprise_edition,report,source=agent");
```

`source=agent` means this is defined in the agent (in the C code), as opposed to being defined in the policy.